bundles/fr.adaussy.permadeler.model.design/src/fr/adaussy/permadeler/model/design/services/ValidationService.java

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Arthur Daussy.
+ *
+ * This program and the accompanying materials are made 
+ * available under the terms of the Eclipse Public License 2.0 
+ * which is available at https://www.eclipse.org/legal/epl-2.0/ 
+ * Contributors:
+ * Lucas Chaigne - initial API and implementation.
+ ******************************************************************************/
+package fr.adaussy.permadeler.model.design.services;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.emf.common.util.EList;
+
+import fr.adaussy.permadeler.model.Permadeler.Area;
+import fr.adaussy.permadeler.model.Permadeler.CompatibilityLink;
+import fr.adaussy.permadeler.model.Permadeler.Genus;
+import fr.adaussy.permadeler.model.Permadeler.Root;
+import fr.adaussy.permadeler.model.utils.EMFUtils;
+
+public class ValidationService {
+
+	public static boolean validatePlantationCompatibility(Area self) {
```

Comment:
Please add java doc. Didn't you get a warning from checkstyle?

Reply:
done